src/ape_ethereum/ecosystem.py

```diff
@@ -551,6 +551,11 @@ def decode_block(self, data: Dict) -> BlockAPI:
         if "transactions" in data:
             data["num_transactions"] = len(data["transactions"])
 
+        if "size" not in data:
+            # NOTE: Due to an issue with `eth_subscribe:newHeads` on Infura
```

Comment:
Question: How can we re-design block API to handle a transaction-less state better.

Reply:
maybe combo of `_size: Optional[int] = Field(None, alias="size")` and a property `size(self) -> int` that either returns the cached value or triggers another call to fetch `.transactions` in order to compute the size from them